packages/service/src/plugin.ts

```diff
@@ -41,7 +41,7 @@ export class PluginsRunner {
     return this;
   }
 
-  async run(fileNames: string[], logger?: PluginLogger): Promise<void> {
```

Comment:
What is the use-case for keeping run? Why not have the generator as the default and when we don’t need to process the yielded file just drain the entire process.

Reply:
@lynchbomb 

Two reasons to keep `run` while writing these:

1. I though the `setImmediate(next)` in the old `run` would still be a little little faster than just yield, when we just want to run all of them at once.
2. `await runAll(fileNames)` looks a little cleaner than `for await (const _ of run(fileName)) { // do thing }`

But I'm totally open to just keeping the generator version, since the difference is just too little.

Reply:
@lynchbomb let me know. If you think it's better to just keep the generator version `run`, I'm happy to change to that.

Reply:
I, the opposite, would keep only `run` with Promise output with a callback and hide the generator inside, like this (example with migrate):

Instead of
```
for await (const f of migrate(input)) {
   migratedFiles.push(f);
}
```
do 
```
await migrate(input, (file) => migratedFiles.push(file))
```

If we want to keep the generator as output - I think it's better to keep only `run => AsyncGenerator` to avoid confusion... but I don't like the idea that devs need to handle yields just by running plugins (`for await (const _ of run(fileName)) { // do thing }`).